compiler/src/iree/compiler/Codegen/Common/FissionTransferOpsInControlFlow.cpp

```diff
@@ -216,6 +215,14 @@ static bool isLegal(vector::TransferReadOp readOp,
           cast<MemRefType>(writeOp.getBase().getType()))) {
     return false;
   }
+
+  // Only the read/write ops may have side effects, we assume we can
+  // re-order/erase the other ops freely.
+  if (!llvm::all_of(forOp.getOps(), [&](Operation &op) -> bool {
```

Comment:
I think this might be more efficient as a separate function called after we find the fisson target, so we dont keep walking all the ops in the scf.for in a loop.